hybrid/pkg/agent/agent.go

```
@@ -0,0 +1,183 @@
+		newInterceptor := p.interceptor.SpawnInterceptor()
+		newInterceptor.SetPluginName(p.pluginList[i].PluginName)
+		interceptors[i] = newInterceptor
+
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		result := elem.MethodByName("AidAttestation").Call([]reflect.Value{reflect.ValueOf(newInterceptor)})
+		err := result[0].Interface()
+
+		if err != nil {
+			errorString := fmt.Sprintf("%v", err)
+			return status.Errorf(codes.Internal, "An error occurred during AidAttestation of the %v plugin. The error was %v", p.pluginList[i].PluginName, errorString)
+		}
+	}
+
+	combinedMessage := common.PluginMessageList{}
+	for i := 0; i < len(interceptors); i++ {
+		combinedMessage.Messages = append(combinedMessage.Messages, interceptors[i].GetMessage())
+	}
+
+	return p.interceptor.SendCombined(combinedMessage)
+}
+
+func (p *HybridPluginAgent) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	pluginData, configError := p.decodeStringAndTransformToAstNode(req.HclConfiguration)
+
+	if configError != nil {
+		return &configv1.ConfigureResponse{}, configError
+	}
+
+	pluginNames, pluginsData := p.parseReceivedData(pluginData)
+
+	p.pluginList, p.initStatus = p.initPlugins(pluginNames)
+
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return nil, p.initStatus
+	}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		req.HclConfiguration = pluginsData[p.pluginList[i].PluginName]
+
+		methodCall := elem.MethodByName("Configure")
+
+		if methodCall.Kind() == 0 {
+			return &configv1.ConfigureResponse{}, status.Errorf(codes.Internal, "Error configuring plugin %v.", p.pluginList[i].PluginName)
+		}
+
+		result := methodCall.Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(req)})
+		err := result[1]
+
+		if !err.IsNil() {
+			return &configv1.ConfigureResponse{}, status.Errorf(codes.Internal, "Error configuring one of the supplied plugins.")
```

errors is lost here, I think you must add it to the chain

---

done